runtimes/native/src/framebuffer.c

```diff
@@ -317,38 +317,28 @@ void w4_framebufferTextUtf16 (const uint16_t* str, int byteLength, int x, int y)
 void w4_framebufferBlit (const uint8_t* sprite, int dstX, int dstY, int width, int height,
     int srcX, int srcY, int srcStride, bool bpp2, bool flipX, bool flipY, bool rotate) {
 
-    int clipXMin = w4_max(0, dstX) - dstX;
-    int clipYMin = w4_max(0, dstY) - dstY;
-    int clipXMax = w4_min(width, WIDTH - dstX);
-    int clipYMax = w4_min(height, HEIGHT - dstY);
     uint16_t colors = *drawColors;
 
     if (rotate) {
         flipX = !flipX;
     }
 
-    for (int row = clipYMin; row < clipYMax; ++row) {
-        for (int col = clipXMin; col < clipXMax; ++col) {
-            // Determine the local position on the sprite
-            int sx, sy;
-            if (rotate) {
-                sx = row;
-                sy = col;
-            } else {
-                sx = col;
-                sy = row;
-            }
-            if (flipX) {
-                sx = clipXMax - sx - 1;
-            }
-            if (flipY) {
-                sy = clipYMax - sy - 1;
+    for (int y = 0; y < height; y++) {
+        for (int x = 0; x < width; x++) {
+            // Calculate sprite target coords
+            const int tx = dstX + (rotate ? y : x);
+            const int ty = dstY + (rotate ? x : y);
+            if (tx < 0 || ty < 0 || tx >= WIDTH || ty >= HEIGHT) {
+                continue;
```

Comment:
Instead of doing a clipping check on every pixel, I wonder if it would it be possible to keep it as before where we precalculate the clipped rectangle?